tests/measurements/test_expval.py

```diff
@@ -19,28 +19,60 @@
 from pennylane.measurements import Expectation
 
 
+# TODO: Remove this when new CustomMP are the default
+def custom_measurement_process(device, spy):
+
+    assert len(spy.call_args_list) > 0  # make sure method is mocked properly
+
+    samples = device._samples  # pylint: disable=protected-access
+    state = device._state  # pylint: disable=protected-access
+    call_args_list = list(spy.call_args_list)
+    for call_args in call_args_list:
+        obs = call_args.args[1]
+        shot_range, bin_size = (
+            call_args.kwargs["shot_range"],
+            call_args.kwargs["bin_size"],
+        )
+        # no need to use op, because the observable has already been applied to ``self.dev._state``
+        meas = qml.expval(op=obs)
+        old_res = device.expval(obs, shot_range=shot_range, bin_size=bin_size)
+        if device.shots is None:
+            new_res = meas.process_state(state=state, wire_order=device.wires)
+        else:
+            new_res = meas.process_samples(
+                samples=samples, wire_order=device.wires, shot_range=shot_range, bin_size=bin_size
+            )
+        assert qml.math.allequal(old_res, new_res)
+
+
 class TestExpval:
     """Tests for the expval function"""
 
+    @pytest.mark.parametrize("shots", [None, 1000, [1000, 10000]])
     @pytest.mark.parametrize("r_dtype", [np.float32, np.float64])
-    def test_value(self, tol, r_dtype):
+    def test_value(self, r_dtype, mocker, shots):
         """Test that the expval interface works"""
-        dev = qml.device("default.qubit", wires=2)
+        dev = qml.device("default.qubit", wires=2, shots=shots)
         dev.R_DTYPE = r_dtype
 
         @qml.qnode(dev, diff_method="parameter-shift")
         def circuit(x):
             qml.RX(x, wires=0)
             return qml.expval(qml.PauliY(0))
 
+        new_dev = circuit.device
```

Comment:
Why do we need to define this "new" device? How is it different from the old one?

Reply:
When instantiating the `QNode` object, the `circuit.device` object might be replaced by a new device object. For example, when using an interface, we replace it with the device specific to that interface.

Reply:
Any reason none of the previous custom MPs require this?